Tests/Common/Background Queue/QueueIntegrationTest.swift

```diff
@@ -17,60 +17,80 @@ class QueueIntegrationTest: IntegrationTest {
 
     #if !os(Linux) // LINUX_DISABLE_FILEMANAGER
     func test_addTask_expectSuccessfullyAdded() {
-        let addTaskActual = queue.addTask(
+        let expect = expectation(description: "Expected to finish async operation")
+
+        queue.addTask(
             type: String.random,
             data: ["foo": "bar"],
             groupStart: .identifiedProfile(identifier: String.random),
             blockingGroups: [.identifiedProfile(identifier: String.random)]
-        )
-        XCTAssertTrue(addTaskActual.success)
-        XCTAssertEqual(addTaskActual.queueStatus.numTasksInQueue, 1)
+        ) { addTaskActual in
+            XCTAssertTrue(addTaskActual.success)
+            XCTAssertEqual(addTaskActual.queueStatus.numTasksInQueue, 1)
+
+            expect.fulfill()
+        }
+
+        waitForExpectations()
     }
 
     func test_addTaskThenRun_expectToRunTaskInQueueAndCallCallback() {
         httpRequestRunnerStub.queueSuccessfulResponse()
 
-        _ = queue.addTask(
+        let expectToAddTask = expectation(description: "Expected to add task")
+        queue.addTask(
             type: QueueTaskType.identifyProfile.rawValue,
             data: IdentifyProfileQueueTaskData(identifier: String.random, attributesJsonString: nil),
             groupStart: .identifiedProfile(identifier: String.random),
             blockingGroups: [.identifiedProfile(identifier: String.random)]
-        )
+        ) { _ in
+            expectToAddTask.fulfill()
+        }
 
-        let expect = expectation(description: "Expect to complete")
+        waitForExpectations()
```

Comment:
To assert that calling `queue.addTask` adds an entry to the BQ? 

Sure, I see value in that kind of test. I don't agree that this check should go into this test function as this test function's purpose is designed around `queue.run` instead of `queue.addTask`. However, I am adding this check to the test function `test_addTask_expectSuccessfullyAdded`.